personaelabs/nym
# Sentry setup

Add the Sentry sdk to frontend/. Both server-side and client-side errors will be reported to our Sentry workspace.

### Privacy concerns
I tried configuring the SDK NOT to send the client information (browser, device info) as much as possible. But there could still be some things sent that could lead to reducing anonymity. (For example, in an extreme case, a nymNam and an ETH address are sent in a single error message).
I considered not using any error reporting SDK as well due to its possible privacy-breaking features, but I think we can experiment with it during development and see what it could leak.

### How to test
Replace `packages/frontend/src/pages/index.tsx` with the following to test reporting an error from the frontend. I invited everyone on the team to the Sentry workspace, so you should be able to see the error pop up in the Sentry dashboard.

```
import Posts from '@/components/Posts';
import { ErrorBoundary } from 'react-error-boundary';

export default function Home() {
  return (
    <ErrorBoundary fallback={<div>Top level error</div>}>
      <button
        onClick={() => {
          throw new Error('I crashed!');
        }}
        className="text-2xl"
      >
        Break the world
      </button>
      <Posts />
    </ErrorBoundary>
  );
}

```
